pkg/controller/operators/catalog/operator.go

```diff
@@ -1140,6 +1152,14 @@ func (o *Operator) syncInstallPlans(obj interface{}) (syncError error) {
 			return
 		}
 
+		if !reflect.DeepEqual(plan.Spec.ClusterServiceVersionNames, out.Spec.ClusterServiceVersionNames) {
```

Comment:
Updating the spec here, followed by updating the status right below, will be a source of errors/flakes (i.e. we have to be sure to handle the case where updating the spec errors before updating the status).